### PR TITLE
Automatically create Issue on /bug

### PR DESCRIPTION
**Pull request by Byteingpython**
## Checklist

 - [ x ] Read [Contribution Guidelines](https://github.com/SlimeCloud/java-SlimeBot#contributing)
 - [ x ] Tested the Code
 - [ x ] This PR is ready to review and merge

## Description
Wenn /bug ausgeführt wird, wird automatisch ein Github Issue erstellt

## Related Issue
#52

### CODE REVIEW

File: src/main/java/com/slimebot/commands/BugCommand.java [hunk 1]
```diff
@@ -2,69 +2,213 @@
...
-					)
-					.setActionRow(Button.secondary("bug:close", "Bug schließen")).queue();
-		}));
-
-		manager.getEventManager().registerHandler(new ButtonHandler("bug:close", event -> {
-			event.getMessage().delete().queue();
-			event.reply("Der Bug wurde erfolgreich geschlossen!").setEphemeral(true).queue();
-		}));
-	}
+public class BugCommand extends ListenerAdapter {
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/commands/BugCommand.java:28]: Commands should not be Listeners

File: src/main/java/com/slimebot/commands/BugCommand.java [hunk 2]
```diff
@@ -2,69 +2,213 @@
...
+                    """);
+            config.save();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @ApplicationCommandMethod()
+    public void performCommand(SlashCommandInteractionEvent event) {
+        if (!event.getFullCommandName().equals("bug")) return;
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/commands/BugCommand.java:97]: You don't have to manually check for that, DiscordUtils does that for you

File: src/main/java/com/slimebot/commands/BugCommand.java [hunk 3]
```diff
@@ -2,69 +2,213 @@
...
+                    %expected% = Expected behavior
+                    %solution% = Solution idea
+                    """);
+            config.save();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @ApplicationCommandMethod()
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/commands/BugCommand.java:95]: You don't need empty parenthesis

File: src/main/java/com/slimebot/commands/BugCommand.java [hunk 4]
```diff
@@ -2,69 +2,213 @@
...
+                .addActionRow(howToReproduce)
+                .addActionRow(expected)
+                .addActionRow(solution)
+                .build();
+
+        event.replyModal(modal).queue();
+    }
+
+    @Override
+    public void onModalInteraction(ModalInteractionEvent event) {
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/commands/BugCommand.java:147]: Use a ModalHandler in a `@WhenFinished` `public void setup(DiscordUtils manager)` method. See how it is done [here](https://github.com/MineKing9534/java-SlimeBot/blob/75e008626725c18d28cb50bce3aac69e50811bbf/src/main/java/com/slimebot/commands/FdmdsCommand.java#L58C3-L58C3).

File: src/main/java/com/slimebot/main/Main.java [hunk 5]
```diff
@@ -101,7 +101,8 @@ public static void main(String[] args) throws IOException {
 				.addEventListeners(new StartupListener())
 				.addEventListeners(new TimeoutListener())
 				.addEventListeners(new JoinListener())
-				.addEventListeners(new StaffMessage());
+				.addEventListeners(new StaffMessage())
+				.addEventListeners(new BugCommand());
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/main/Main.java:105]: Don't add the command as listener but use `registerCommand(BugCommand.class)` instead

Comment (Byteingpython) [reply src/main/java/com/slimebot/main/Main.java]: Done

File: src/main/java/com/slimebot/commands/BugCommand.java (hunk 4 above)
Comment (Byteingpython) [reply src/main/java/com/slimebot/commands/BugCommand.java]: Done

File: src/main/java/com/slimebot/commands/BugCommand.java (hunk 3 above)
Comment (Byteingpython) [reply src/main/java/com/slimebot/commands/BugCommand.java]: Done

File: src/main/java/com/slimebot/commands/BugCommand.java (hunk 2 above)
Comment (Byteingpython) [reply src/main/java/com/slimebot/commands/BugCommand.java]: Done

File: src/main/java/com/slimebot/commands/BugCommand.java [hunk 6]
```diff
@@ -6,65 +6,214 @@
...
+        event.replyModal(modal).queue();
+    }
+
+    @WhenFinished
+    public void setup(DiscordUtils utils){
+        utils.getEventManager().registerHandler(new ModalHandler("bug", this::processModal));
+    }
+
+    public void processModal(ModalInteractionEvent event) {
+        if (!event.getModalId().equals("bug")) return;
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/commands/BugCommand.java:153]: The ModalHandler already checks for that

Comment (Byteingpython) [reply src/main/java/com/slimebot/commands/BugCommand.java]: Done

**Review by MineKing9534 — APPROVED** [review]

**Comment by MineKing9534** [comment]
#71

**Comment by MatitoDA** [comment]
Merged in #71

**Pull request closed without merging**